### ISSUE TITLE
MAF liftover misaligns tabs (incorrectly removes consecutive tabs)

### ISSUE THREAD

**Issue by jfx319** [liguowang/CrossMap#12]
I am observing a behavior during maf -> maf genome assembly conversion, where consecutive tabs in the input maf are being removed in the output maf, which misaligns the columns. 

Seems like it should be an easy fix... any advice?

#### Visualize the difference:  
- Input:  https://gist.github.com/jfx319/4f09b2619643b18d6958fd83f3f092bf
- Output:  https://gist.github.com/jfx319/2bbce1955ca658d5a7d08beec8e5292c

#### Exact files used in CrossMap v0.3.9:
```bash
#input example maf (named as tsv for github auto-format visualization)
wget https://gist.githubusercontent.com/jfx319/4f09b2619643b18d6958fd83f3f092bf/raw/020d2d42dd4a56c81be76fe75ff67cb9508835cf/foo.GRCh37.maf.tsv

#output example maf
wget https://gist.githubusercontent.com/jfx319/2bbce1955ca658d5a7d08beec8e5292c/raw/4fc2e5f723a0b97207bbbe627fbc3c05058766fe/foo.GRCh38.crossmap.maf.tsv

#target genome
wget ftp://ftp.ensembl.org/pub/release-93/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
gunzip Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
md5sum Homo_sapiens.GRCh38.dna.primary_assembly.fa
    #9e6b9465dc708d92bf6d67e9c9fa9389

#chain file
wget ftp://ftp.ensembl.org/pub/assembly_mapping/homo_sapiens/GRCh37_to_GRCh38.chain.gz
md5sum GRCh37_to_GRCh38.chain.gz
    #2a862e70fa1afc08f1f3e7a8cc5cda14

#########################################

CrossMap.py
#Program: CrossMap (v0.3.9)

CrossMap.py \
    maf \
    GRCh37_to_GRCh38.chain.gz \
    foo.GRCh37.maf \
    Homo_sapiens.GRCh38.dna.primary_assembly.fa \
    GRCh38 \
    foo.GRCh38.maf
#@ 2020-03-24 04:23:11: Read the chain file:  GRCh37_to_GRCh38.chain.gz
#@ 2020-03-24 04:23:11: Lifting over ...
#@ 2020-03-24 04:23:11: Total entries: 4
#@ 2020-03-24 04:23:11: Failed to map: 0
```

Note: I don't think this is contributing to the issue, but for full disclosure, I am running from this [container](https://hub.docker.com/r/miguelpmachado/crossmap/dockerfile):
```
docker run -it --rm -v '/scratch:/mnt/scratch' miguelpmachado/crossmap:0.3.9 bash
```

**Comment by jfx319** [comment]
I suspect it has to do with not specifying an explicit delimiter to python's `str.split()` method, which triggers a different algorithm:
- https://stackoverflow.com/a/51468354

I suggest changing [line 682](https://github.com/liguowang/CrossMap/blob/2f5f050532766d58a4563e3dc0c96c2d25eb5e90/bin/CrossMap.py#L682),  
- from 
`fields = str.split(line)` 
- to `fields = str.split(line, sep = '\t')`
should do the trick.

Working backwards:
```python
f = open('foo.GRCh37.maf', 'rb')
line1 = f.readline()
line2 = f.readline()
f.close()

line1 = line1.decode('utf8').strip().replace("\r", "") #per lib/cmmodule/ireader.py
fields1 = str.split(line1)
len(fields1)
#134

line2 = line2.decode('utf8').strip().replace("\r", "")
fields2 = str.split(line2)
len(fields2)
#65

fields2_explicit_split = str.split(line2, sep='\t')
len(fields2_explicit_split)
#134
```

Will file a pull simple pull request.

**Comment by liguowang** [comment]
Hi Jeff,

We have released a new version (v0.4.1) to address this issue.  To upgrade,
run:

pip3 install crossmap --upgrade

please let me know if you have other questions.

thanks!

Liguo

**Comment by jfx319** [comment]
Thanks!

I can verify that 43b935bf557707d333eee88c62eb99c3e35d898a fixed it for my MAF file conversion. 

If consecutive tabs happen in other file formats (bed, vcf, etc), then I suppose someone will raise a separate issue.
